Automattic/wp-calypso
# composite-checkout: Remove custom localize module

#### Changes proposed in this Pull Request

This removes the `useLocalize` hook and the `LocalizeProvider` from the `@automattic/composite-checkout` package. Instead, it uses `__` from `@wordpress/i18n` (which we were already using under the hood) and `@automattic/react-i18n`.

This is part of https://github.com/Automattic/wp-calypso/issues/42831

#### Testing instructions

This should not cause any changes. Load composite checkout and be sure the strings look ok (they will not be translated yet).
